src/cdk-experimental/scrolling/virtual-scroll-viewport.spec.ts

```diff
@@ -1,46 +1,394 @@
+        triggerScroll(viewport, offset);
+
+        const expectedRange = {
+          start: Math.floor(offset / testComponent.itemSize),
+          end: Math.floor(offset / testComponent.itemSize) + 4
+        };
+        expect(viewport.getRenderedRange()).toEqual(expectedRange);
+        expect(viewport.getOffsetToRenderedContentStart())
+            .toBe(expectedRange.start * testComponent.itemSize);
+        expect(viewport.measureRenderedContentSize()).toBe(testComponent.viewportSize);
+      }
+    }));
+
+    it('should update viewport as user scrolls left', fakeAsync(() => {
+      finishInit(fixture);
+
+      const maxOffset =
+          testComponent.itemSize * testComponent.items.length - testComponent.viewportSize;
+      for (let offset = maxOffset; offset >= 0; offset -= 10) {
+        triggerScroll(viewport, offset);
+
+        const expectedRange = {
+          start: Math.floor(offset / testComponent.itemSize),
+          end: Math.floor(offset / testComponent.itemSize) + 4
+        };
+        expect(viewport.getRenderedRange()).toEqual(expectedRange);
+        expect(viewport.getOffsetToRenderedContentStart())
+            .toBe(expectedRange.start * testComponent.itemSize);
+        expect(viewport.measureRenderedContentSize()).toBe(testComponent.viewportSize);
+      }
+    }));
+  });
 });
 
+
+/** Finish initializing the virtual scroll component at the beginning of a test. */
+function finishInit(fixture: ComponentFixture<any>) {
+  // One the first cycle we render and measure the viewport.
+  fixture.detectChanges();
+  flush();
+
+  // On the second cycle we render the items.
+  fixture.detectChanges();
+  flush();
+}
+
+/** Trigger a scroll event on the viewport (optionally setting a new scroll offset). */
+function triggerScroll(viewport: CdkVirtualScrollViewport, offset?: number) {
+  if (offset !== undefined) {
+    if (viewport.orientation == 'horizontal') {
+      viewport.elementRef.nativeElement.scrollLeft = offset;
+    } else {
+      viewport.elementRef.nativeElement.scrollTop = offset;
+    }
+  }
+  dispatchFakeEvent(viewport.elementRef.nativeElement, 'scroll');
+  animationFrameScheduler.flush();
+}
+
+
+@Component({
+  template: `
+    <cdk-virtual-scroll-viewport
+        class="viewport" [itemSize]="itemSize" [bufferSize]="bufferSize"
+        [style.height.px]="viewportSize" [style.width.px]="viewportCrossSize">
+      <div class="item" *cdkVirtualFor="let item of items; let i = index"
+           [style.height.px]="itemSize">
+        {{i}}
+      </div>
+    </cdk-virtual-scroll-viewport>
+  `,
+  styles: [`.cdk-virtual-scroll-content-wrapper { display: flex; flex-direction: column; }`],
+  encapsulation: ViewEncapsulation.None,
+})
+class FixedVirtualScroll {
+  @ViewChild(CdkVirtualScrollViewport) viewport: CdkVirtualScrollViewport;
+  @ViewChild(CdkVirtualForOf, {read: ViewContainerRef}) cdkForOfViewContainer: ViewContainerRef;
+
+  @Input() viewportSize = 200;
```

Comment:
These aren't really inputs.

Reply:
They're not inputs to the viewport, but they are inputs to my test component. I will want to create tests that check that the viewport handles re-rendering its content on size change once I add that feature